### PR TITLE
Register channels to ResetLoggersWorkerSubscriber

### PR DESCRIPTION
**Pull request by l-vo**
Channels registration for symfony/symfony#40761

### CODE REVIEW

**Comment by fabpot** [comment]
The related PR in Symfony has been merged for 5.3.

File: MonologBundle.php [hunk 1]
```diff
@@ -38,6 +41,10 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new FixEmptyLoggerPass($channelPass));
         $container->addCompilerPass(new AddProcessorsPass());
         $container->addCompilerPass(new AddSwiftMailerTransportPass());
+
+        if (class_exists(MessageBusInterface::class) && class_exists(ResetLoggersWorkerSubscriber::class)) {
```
Comment (chalasr) [inline MonologBundle.php:45]: Only ResetLoggersWorkerSubscribed is needed, right?

Comment (l-vo) [reply MonologBundle.php:45]: No, both are needed to me. `ResetLoggersWorkerSubscriber` is from MonologBridge. If Monolog bridge is installed but not Messenger, we don't want `ResetLoggersWorkerSubscriber` to be activated.

Comment (chalasr) [reply MonologBundle.php:45]: Of course, thanks.

**Review by chalasr — APPROVED** [review]

**Review by lyrixx — APPROVED** [review]

**Comment by alxndrbauer** [comment]
Is there any particular reason why this is not merged yet?

**Comment by lyrixx** [comment]
I don't thin this is good actually. The listener will always be registered. That means when Symfony handle an http request, if a message is sent in a sync transport, monolog will be flushed. So if something goes wrong after that, all log about routing, security etc will be lost 😐

I created another pr to fix this issue more "globally" https://github.com/symfony/symfony/pull/41163

**Comment by lyrixx** [comment]
@l-vo Thanks for working on this topic.

As I have explain it in my previous comment, the alternative implementation provided by symfony/symfony#41163  (and subsequent PR) is better.

I'm also [deprecating](https://github.com/symfony/symfony/issues/43271) the integration in the bridge

I hope your understand the move here.

So, I'm sorry for closing your PR, but I think it's better like this.

Again, thanks for your first PR.

**Comment by l-vo** [comment]
@lyrixx No problem, the reason is legit, thank you 🙂

**Pull request closed without merging**